source/library/CabalGild/Unstable/Main.hs

```diff
@@ -65,7 +71,14 @@ mainWith ::
 mainWith arguments = do
   flags <- Flag.fromArguments arguments
   config <- Config.fromFlags flags
-  context <- Context.fromConfig config
+  context <- do
+    context <- Context.fromConfig config
```

Comment:
Rather than updating the context, this could update the config by using `applyFlag config (Flag.IO f)`. That way you wouldn't need `Context.setIO`. 

Alternatively, this logic could be pushed into `Context.fromConfig`.